moveit_ros/hybrid_planning/hybrid_planning_manager/hybrid_planning_manager_component/src/hybrid_planning_manager.cpp

```diff
@@ -147,16 +147,46 @@ bool HybridPlanningManager::initialize()
   hybrid_planning_request_server_ = rclcpp_action::create_server<moveit_msgs::action::HybridPlanner>(
       this->get_node_base_interface(), this->get_node_clock_interface(), this->get_node_logging_interface(),
       this->get_node_waitables_interface(), hybrid_planning_action_name,
+      // Goal callback
       [](const rclcpp_action::GoalUUID& /*unused*/,
          std::shared_ptr<const moveit_msgs::action::HybridPlanner::Goal> /*unused*/) {
         RCLCPP_INFO(LOGGER, "Received goal request");
         return rclcpp_action::GoalResponse::ACCEPT_AND_EXECUTE;
       },
-      [](const std::shared_ptr<rclcpp_action::ServerGoalHandle<moveit_msgs::action::HybridPlanner>>& /*unused*/) {
+      // Cancel callback
+      [this](std::shared_ptr<rclcpp_action::ServerGoalHandle<moveit_msgs::action::HybridPlanner>> /*unused*/) {
+        // Prevent any new global or local requests from going out
+        stop_hybrid_planning_ = true;
```

Comment:
Just saw this now... Shouldn't this use the API of the ServerGoalHandle to flag the goal as canceled, so that the affected code can use `ServerGoalHandle::is_canceling()`? https://docs.ros2.org/latest/api/rclcpp_action/classrclcpp__action_1_1ServerGoalHandle.html

Reply:
If I understand, you would like me to add something like this, here in the cancel callback:

        `hybrid_planning_goal_handle->canceled();`
        
Looking at the [tutorial](https://docs.ros.org/en/foxy/Tutorials/Actions/Writing-a-Cpp-Action-Server-Client.html), they do not use canceled() within the cancel callback. My guess is, `goal->canceled()` would be redundant when used like this.

Reply:
I can follow up with a new PR if I missed something here.